firefox-ios/Client/Frontend/Translations/Service/TranslationsService.swift

```diff
@@ -0,0 +1,158 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+import WebKit
+
+/// Errors thrown by `TranslationsService` when preconditions or WebView state are invalid.
+enum TranslationsServiceError: Error, Equatable {
+    case missingWebView
+    case missingSourceLanguage
+    case missingDeviceLanguage
+    case jsEvaluationFailed(reason: String)
+}
+
+/// Default implementation of `TranslationsServiceProtocol`
+@MainActor
+final class TranslationsService: TranslationsServiceProtocol {
+    private let windowManager: WindowManager
+    private let languageDetector: LanguageDetectorProvider
+    private let modelsFetcher: TranslationModelsFetcherProtocol
+    private let translationsEngine: TranslationsEngine
+    private let logger: Logger
+
+    init(
+        windowManager: WindowManager = AppContainer.shared.resolve(),
```

Comment:
We can do this in a follow up refactor, but the window manager seems very specific to our implementation, could we do a method injection on the selected web view instead? We wouldn't also need to know about the windowUUID

Instead of using 

```
 let webView = try currentWebView(for: windowUUID)
 ```
 maybe we can inject it instead?
 ```
 func shouldOfferTranslation(webView, WebView, for windowUUID: WindowUUID)
 ```